File: WebApp_API/WebApp_API/Models/Product.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace WebApp_API.Models
{
    public class Product
    {
        [Key] public int Id { get; set; }
        [Required, MaxLength(300)] public string Name { get; set; }
        [Required, MaxLength(300)] public string Slug { get; set; }
        [MaxLength(1000)] public string? ShortDescription { get; set; }
        public string? Description { get; set; }
        [Column(TypeName = "decimal(18,2)")] public decimal Price { get; set; }
        [MaxLength(1000)] public string? ImageUrl { get; set; }
        public int? CategoryId { get; set; }
        [ForeignKey("CategoryId")] public Category? Category { get; set; }
    }
}

File: reactts_frontend/src/types/models/Product.ts
import type { Category } from "./Category";

export interface Product {
  id: number | string;
  name: string;
  slug: string;
  price: number;
  description?: string;
  shortDescription?: string;
  imageUrl?: string;
  images: string[];
  categoryId?: number | string;
  category?: Category;
  options?: Array<{
    optionName: string;
    value: string;
  }>;
  selectedOptionValueIds?: Array<number>;
}
